src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/volume/v2/OpenStackVolumePluginTest.java

```diff
@@ -1,201 +1,524 @@
+    
+    private static final int ERROR_STATUS_CODE = 500;
+    
+    private OpenStackVolumePlugin plugin;
     private OpenStackHttpClient client;
 
     @Before
     public void setUp() throws Exception {
-        PropertiesHolder propertiesHolder = PropertiesHolder.getInstance();
-        Properties properties = propertiesHolder.getProperties();
-        properties.put(OpenStackVolumePlugin.VOLUME_NOVAV2_URL_KEY, FAKE_STORAGE_URL);
-        String cloudConfPath = HomeDir.getPath() + SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME + File.separator
-                + "default" + File.separator + SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
-        this.openStackVolumePlugin = Mockito.spy(new OpenStackVolumePlugin(cloudConfPath));
+        this.testUtils.mockReadOrdersFromDataBase();
         this.client = Mockito.mock(OpenStackHttpClient.class);
-        this.openStackVolumePlugin.setClient(this.client);
-        this.openStackV3Token = new OpenStackV3User(FAKE_USER_ID, FAKE_NAME, FAKE_TOKEN_VALUE, FAKE_PROJECT_ID);
-    }
 
-    // test case: Check if the request in requestInstance() is executed properly with the right parameters.
+        String openstackCloudConfPath = HomeDir.getPath() 
+                + SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME
+                + File.separator 
+                + TestUtils.DEFAULT_CLOUD_NAME 
+                + File.separator
+                + SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
+
+        this.plugin = Mockito.spy(new OpenStackVolumePlugin(openstackCloudConfPath));
+        this.plugin.setClient(this.client);
+    }
+    
+    // test case: When calling the isReady method with the cloud states available,
+    // this means that the state of volume is READY and it must return true.
     @Test
-    public void testRequestInstance() throws FogbowException, HttpResponseException {
+    public void testIsReadySuccessful() {
         // set up
-        VolumeOrder volumeOrder = Mockito.mock(VolumeOrder.class);
-
-        Mockito.doReturn(FAKE_VOLUME_JSON).when(this.client).doPostRequest(
-                Mockito.anyString(), Mockito.any(), Mockito.any(OpenStackV3User.class));
+        String cloudState = OpenStackStateMapper.AVAILABLE_STATUS;
 
         // exercise
-        String instanceString = this.openStackVolumePlugin.requestInstance(volumeOrder, this.openStackV3Token);
+        boolean status = this.plugin.isReady(cloudState);
 
         // verify
-        Mockito.verify(this.client).doPostRequest(Mockito.anyString(), Mockito.any(), Mockito.any(OpenStackV3User.class)
-        );
-        Assert.assertEquals(FAKE_VOLUME_ID, instanceString);
+        Assert.assertTrue(status);
+    }
+    
+    // test case: When calling the isReady method with the cloud states different
+    // than available, this means that the state of volume is not READY and it must
+    // return false.
+    @Test
+    public void testIsReadyUnsuccessful() {
```

Comment:
this is not an unsuccessful test case. The method worked as expected and correctly. Maybe u should rename it

Reply:
check the same thing for the other method

Reply:
You're right, I'll rename these test methods.